app/cdap/components/Snackbar/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { green, red } from '@material-ui/core/colors';
+import Snackbar from '@material-ui/core/Snackbar';
+import Transition from 'components/Snackbar/Components/Transition/index';
+import { ISnackbarProps } from 'components/Snackbar/types';
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+
+const SnackbarWrapper = styled(Snackbar)`
+  border-radius: 4px;
+  width: 100%;
+  top: 48px !important;
+  background-color: ${(props) => (props.isSuccess ? green[600] : red[600])};
+  padding: 15px 18px 14px 18px;
+  display: block;
+  min-height: 76px;
+  left: 0;
+  z-index: 9;
+`;
+
+export default function({
+  handleCloseError,
+  description = '',
+  isSuccess,
+  snackbarAction,
+  isOpen,
+  setSnackbarState,
+}: ISnackbarProps) {
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setSnackbarState(false);
+      handleCloseError();
+    }, 5000);
+    return () => {
+      setSnackbarState(false);
+      clearTimeout(timer);
+    };
+  }, []);
+
+  const handleClose = () => {
+    setSnackbarState(false);
+    handleCloseError();
+  };
+
+  return (
+    <SnackbarWrapper
```

Comment:
why is this whole component conditionally rendered? Doesn't the mui snackbar conditionally render it? it doesn't look like they conditionally render the snackbar in their examples, they just pass a prop to it https://mui.com/material-ui/react-snackbar/

Reply:
updated the code